rpmpatch/helpers.py

```diff
@@ -188,19 +191,78 @@ def rpmpatch(configdir, srpm, changelog_user, verbose=False):
             # done with this, clean it up
             shutil.rmtree(tempdir)
             if verbose:
-                print '## added un-used items'
+                print('## added un-used items')
 
     if verbose:
-        print '## building rpm'
+        print('## building rpm')
 
     defines_dict = {}
     for thing in cfg_file['define']:
         defines_dict[thing[0]] = thing[1]
 
+    if keep_dist:
+        dist = __guess_srpm_dist(release) 
+        defines_dict['%dist'] = f'.{dist}'
+
     built = specfile.build(defines_dict, cfg_file['program']['compile'],
                            cfg_file['program']['build_targets'])
     return built
 
+
+def __guess_srpm_dist(release):
+    """
+    the dist tag must be extracted from the release. The problem is that it might look like this:
+    - %release.%dist
+    - %release.%dist.%subrelease
+    - %dist.%release
+    so the easiest thing is going through release, mark the index of first
+    not numerical and not dot character then find next dot or end of the
+    string.
+    This method still won't work if %release is something like:
+    git_123123123.el8_6. So it's not very clever
+    There is a separate way to find distag for modular package that is based on
+    the `module+` string.
+    """
+    if 'module+' not in release:
+        # using set to make it a little bit faster than array
+        skip_chars = set(['.']+[str(x) for x in range(0, 9)])
```

Comment:
nie wiem czy powinienem się nad tym zastanawiać minutę co to robi.. Jak już zrozumiałem że to robi po prostu set '. 0 1 2 3 ...' itd to wydaje się banalne. Wolałbym jednak by kod był przejrzysty.

Reply:
fixed